### DISCUSSION TITLE
How can I achieve this UI? child group block `width: fit-content`

### DISCUSSION THREAD

**Discussion by kohheepeace** [WordPress/gutenberg#57934] [Q&A] [answered]
## Problem
I would like to achieve the following "Expected" design. However, I could not find a good method. Any advice would be appreciated.

| Problem | Expected
|--------|--------|
|<img width="1699" alt="スクリーンショット 2024-01-18 3 47 36" src="https://github.com/WordPress/gutenberg/assets/29557494/40d2ebf8-4faf-4a13-878f-141dd8367c22">|<img width="1293" alt="スクリーンショット 2024-01-18 3 32 49" src="https://github.com/WordPress/gutenberg/assets/29557494/c4aa8851-27a9-4f16-87c6-e9a4af082aa1">|

## What I tried
- Add border to `paragraph` block
=> There is no border options for paragraph block (https://github.com/WordPress/gutenberg/issues/43247)

- Wrap `paragraph` with `group` block
=> there is no option "`width: fit-content`"

**Comment by t-hamano** [comment]
Hi @kohheepeace,

Is it possible to achieve the same thing by changing the row layout's "Justification" to "Justify items center" ?

How about changing the Group block to the Stack layout and changing "Justification" to "Justify items center" ?

![stack](https://github.com/WordPress/gutenberg/assets/54422211/23092d60-bff4-47ac-86ec-268c01ec731e)

**Reply by kohheepeace** [reply]
@t-hamano Thanks for your reply.

Unfortunately, it does not work. 

I tried at https://playground.wordpress.net/

| Actual | Expected |
|--------|--------|
|<img width="1710" alt="スクリーンショット 2024-01-18 22 31 46" src="https://github.com/WordPress/gutenberg/assets/29557494/31af0db3-a792-4739-96a3-2c3283910285"> | <img width="1710" alt="スクリーンショット 2024-01-18 22 35 33" src="https://github.com/WordPress/gutenberg/assets/29557494/ed75b5a7-da09-450d-8c2b-892e5d8bc29f"> |

**Reply by t-hamano** [reply]
Are you applying the green background color to the Stack block itself? If you select "Justify items center" for the Stack block, the width of the Stack block itself will not change. The width of the Stack block's _child_ blocks should change depending on the length of its content. It would be easier to understand if you apply a background color to the paragraph block inside the Stack block.

**Reply by kohheepeace** [reply]
My goal is to add **border**. But it is not possible to add border when using `paragraph block`.

So, I used `group block` to wrap paragraph.

Then, another problem is happened. => group block doesn't support `width: fit-content`.

I apologize for my poor explanation. 😢


The following `theme.json` did not add border support to the paragraph block.

```json
  "settings": {
    "appearanceTools": true,
    "border": {
      "color": true,
      "radius": true,
      "style": true,
      "width": true
    },        
    "blocks": {
      "core/paragraph": {
        "border": {
          "radius": true,
          "width": true
        }
      }
    }
  },
  ```

**Reply by t-hamano** [reply, marked as answer]
How about wrapping the paragraph block inside the Stack block with a Group block and apply a background color and border to that Group block?

![stack](https://github.com/WordPress/gutenberg/assets/54422211/da1df106-f2ce-43dc-9bb5-a22f81fc7797)

**Reply by kohheepeace** [reply]
You are a genius! It worked!!! Thanks !!!!
